### PR TITLE
Concurrency thread synchronization / it-academy.by / M-JC1-66-20 / Yauhen Filipau / Done Producer-Consumer, so-so SkyNet

### PR DESCRIPTION
**Pull request by FilippovZ**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

**Comment by FilippovZ** [comment]
the project SkyNet was not fully completed, but due to the large amount of time spent, I would like to receive feedback.  Only 1 thread works in the method "getDetails", and the second Thread enters only after the first is fully completed, at the end. 
I don't know how to solve this problem

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/thread_synchronization/skyNet/Details.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.rakovets.course.java.core.practice.concurrency.thread_synchronization.skyNet;
+
+public enum Details {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/thread_synchronization/skyNet/Details.java:3]:
```suggestion
public enum Detail {
```

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/thread_synchronization/skyNet/Storage.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                storage.remove(0);
+            }
+            if (storage.isEmpty()) {
+                notifyAll();
+            }
+        }
+        notifyAll();
+        return detailsMap;
+    }
+}
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/thread_synchronization/skyNet/Storage.java:65]: Violation SOLID

**Review by rakovets — CHANGES_REQUESTED** [review]
Good job but you must apply design principles.
You need to fix the following problems.

**Pull request closed without merging**